xline/src/server/lease_server.rs

```diff
@@ -286,47 +255,30 @@ where
     ) -> Result<tonic::Response<Self::LeaseKeepAliveStream>, tonic::Status> {
         debug!("Receive LeaseKeepAliveRequest {:?}", request);
         let request_stream = request.into_inner();
-        let stream = if self.is_leader() {
-            self.leader_keep_alive(request_stream).await
-        } else {
-            self.follower_keep_alive(request_stream).await?
+        let stream = loop {
+            if self.lease_storage.is_primary() {
+                break self.leader_keep_alive(request_stream).await;
+            }
+            let leader_id = self.client.get_leader_id().await;
```

Comment:
the `leader_id` returned by `get_leader_id` maybe oudated, because it will read from client's state first, and call this method may have one mord RTT because of `fetch_leader`

Reply:
I leave issue #299  to track this problem. It will be fixed in a new pr